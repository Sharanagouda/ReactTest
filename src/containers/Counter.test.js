import React from "react";
import renderer from "react-test-renderer";
import CounterApp, {Counter, dataReducer} from "./Counter";
import {mount} from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({
    adapter: new Adapter()
});
describe('CounterApp', () => {
    it("SnapShot renders", () =>{
        const component = renderer.create(<CounterApp />);
        let tree =component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Counter", () => {
    test("Snapshot renders", () =>{
        const component = renderer.create(<Counter />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});
const list = ['a', 'b', 'c'];
describe("CounterApp ", () => {
    describe('Reducer', () => {
        it('should set a list ', () => {
            const state = { list: [], error:null };
            const newState = dataReducer(state, {
                type: 'SET_LIST',
                list,
            });
            expect(newState).toEqual({list, error: null});
        });
        it(" should reset the error if list is set",  () => {
            const state = {list: [], error: true};
            const newState = dataReducer(state, {
                type: "SET_LIST",
                list,
            });
            expect(newState).toEqual({list, error:null});
        });
        it("should set the error", () => {
            const state = {list: [], error: null};
            const newState =  dataReducer(state, {
                type: "SET_ERROR",
            });
            expect(newState.error).toBeTruthy();
        });
    });
});

describe("CounterApp", () => {
    it("renders the inner counter component length  ", () => {
        const wrapper = mount(<CounterApp/>);
        expect(wrapper.find(Counter).length).toEqual(1);
    });
    it("passes all props to counter ", () => {
        const wrapper = mount(<CounterApp/>);
        const counterWrapper = wrapper.find(Counter);
        expect(counterWrapper.find('p').text()).toEqual('0');
    })
})