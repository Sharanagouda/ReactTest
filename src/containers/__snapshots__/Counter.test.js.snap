// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot renders 1`] = `
<div>
  <p />
</div>
`;

exports[`CounterApp SnapShot renders 1`] = `
<div>
  <h1>
    My Counter!
  </h1>
  <div>
    <p>
      0
    </p>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
  <h3>
    My Async Data
  </h3>
  <ul />
</div>
`;
